addons/addon-environment-sc-api/packages/environment-sc-workflow-steps/lib/steps/check-launch-dependency/check-launch-dependency.js

```diff
@@ -0,0 +1,433 @@
+   * A method to decide when to resume the workflow.
+   * This method checks for the status of the ALB being provisioned and returns true when the
+   * stack has completed successfully. If the stack encountered any errors, the method throws an
+   * error.
+   *
+   * @returns {Promise<boolean>}
+   */
+    async shouldResumeWorkflow() {
+        const [requestContext, stackId, resolvedVars] = await Promise.all([
+            this.payloadOrConfig.object(inPayloadKeys.requestContext),
+            this.state.string('STACK_ID'),
+            this.payloadOrConfig.object(inPayloadKeys.resolvedVars),
+        ]);
+        const cfn = await this.getCloudFormationService(requestContext, resolvedVars);
+        const stackInfo = (await cfn.describeStacks({ StackName: stackId }).promise()).Stacks[0];
+
+        if (_.includes(failureStatuses, stackInfo.StackStatus)) {
+            // If provisioning failed then throw error, any unhandled workflow errors
+            // are handled in "onFail" method
+            throw new Error(`ALB Stack operation failed with message: ${stackInfo.StackStatusReason}`);
+        }
+
+        if (_.includes(successStatuses, stackInfo.StackStatus)) {
+            // If the provisioning completed successfully then return true to resume workflow}
+            return true;
+        }
+        // Return false to continue waiting for the product provisioning to complete
+        return false;
+    }
+
+    /**
+  * Method to perform tasks after the alb provisioning is completed successfully.
+  * @returns {Promise<*>}
+  */
+    async onSuccessfulCompletion() {
+        const [requestContext, resolvedVars, stackId] = await Promise.all([
+            this.payloadOrConfig.object(inPayloadKeys.requestContext),
+            this.payloadOrConfig.object(inPayloadKeys.resolvedVars),
+            this.state.string('STACK_ID'),
+        ]);
+
+        const cfn = await this.getCloudFormationService(requestContext, resolvedVars);
+        const stackInfo = (await cfn.describeStacks({ StackName: stackId }).promise()).Stacks[0];
+        const stackOutputs = this.getStackOutputs(stackInfo);
+        return await this.handleStackCompletion(stackOutputs);
+    }
+
+    async reportTimeout() {
+        const [stackName, resolvedVars] = await Promise.all([
+            this.state.string('STACK_ID'),
+            this.payloadOrConfig.object(inPayloadKeys.resolvedVars),
+        ]);
+        const envName = resolvedVars.name;
+        throw new Error(
+            `Error provisioning environment "${envName}". The workflow timed-out because the ALB provisioing stack "${stackName}" did not ` +
+            `complete within the timeout period of 15 days.`,
+        );
+    }
+
+    /**
+    * Method to perform tasks upon some error. The method calls "onEnvProvisioningFailure" method on plugins registered for the extension point "env-provisioning"
+    * @returns {Promise<*>}
+    */
+    async onFail(error) {
+        this.printError(error);
```

Comment:
Where is this function defined? Can we log instead?

Reply:
We have followed the same approach as the launch-product step to print any errors. Please let us know if this needs to be changed.